At least staticcheck should fail:
```
fabric/dataclient.go:50:2: var errServiceNotFound is unused (U1000)
fabric/dataclient.go:65:2: field loggedMissingRouteGroups is unused (U1000)
fabric/dataclient.go:71:2: field testIn is unused (U1000)
fabric/dataclient.go:72:2: field testOut is unused (U1000)
fabric/fabric.go:202:9: this value of ok is never used (SA4006)
fabric/fabric.go:357:2: empty branch (SA9003)
fabric/fabrictest/fixtures.go:33:6: type kubeOptionsParser is unused (U1000)
fabric/fabrictest/fixtures.go:133:6: func compileRegexps is unused (U1000)
```

--- a/fabric/dataclient.go
+++ b/fabric/dataclient.go
@@ -47,7 +47,6 @@ const (
 
 var (
 	errResourceNotFound     = errors.New("resource not found")
-	errServiceNotFound      = errors.New("service not found")
 	errAPIServerURLNotFound = errors.New("kubernetes API server URL could not be constructed from env vars")
 	errInvalidCertificate   = errors.New("invalid CA")
 
@@ -61,15 +60,11 @@ type clusterClient struct {
 	tokenProvider secrets.SecretsProvider
 	httpClient    *http.Client
 	apiURL        string
-
-	loggedMissingRouteGroups bool
 }
 
 type FabricDataClient struct {
 	quit          chan struct{}
 	ClusterClient *clusterClient
-	testIn        string
-	testOut       string
 }
 
 type Options struct {

--- a/fabric/fabric.go
+++ b/fabric/fabric.go
@@ -199,7 +199,7 @@ func (fps *FabricPaths) UnmarshalJSON(value []byte) error {
 								continue
 							}
 							for k, v := range staticMap {
-								response.Headers[k], ok = v.(string)
+								response.Headers[k], _ = v.(string)
 							}
 						case "status":
 							l, ok := staticVal.(float64)
@@ -354,35 +354,35 @@ func validateMethod(fm *FabricMethod) error {
 		return err
 	}
 
-	if n := len(fm.Privileges); n == 0 {
-		/*
-				   TODO(sszuecs): ignore, but needs cross check with
-				   team fabric.  this config exists in
-				   "spp-perf-test-brand-service", which seems to be
-				   wrong by reviwing CRD spec description. But also in
-				   valid cases like:
-			 - Another case, which generates routes for all
-			   employees and services that have uid scope in their
-			   token: /hello: get: {}
-			   This is specified in fabricgateway name: smart-product-platform-test-gateway in
-			   namespace: default Are these routes expected or is this an error?
-			   My validation that I translated from CRD description says it's wrong, but maybe the documentation is just outdated.
+	/*
+		if n := len(fm.Privileges); n == 0 {
+					   TODO(sszuecs): ignore, but needs cross check with
+					   team fabric.  this config exists in
+					   "spp-perf-test-brand-service", which seems to be
+					   wrong by reviwing CRD spec description. But also in
+					   valid cases like:
+				 - Another case, which generates routes for all
+				   employees and services that have uid scope in their
+				   token: /hello: get: {}
+				   This is specified in fabricgateway name: smart-product-platform-test-gateway in
+				   namespace: default Are these routes expected or is this an error?
+				   My validation that I translated from CRD description says it's wrong, but maybe the documentation is just outdated.
 
-			   The last one seems to be on purpose, because of other gateways (fabric-event-scheduler-gateway) use this feature to allow /health access:
-			   /health:
-			     get: {}
+				   The last one seems to be on purpose, because of other gateways (fabric-event-scheduler-gateway) use this feature to allow /health access:
+				   /health:
+				     get: {}
 
-			Same for health in setanta-categories-staging and /metrics in setanta-category-gatekeeper
+				Same for health in setanta-categories-staging and /metrics in setanta-category-gatekeeper
 
-			/api/brands/*:
-			  delete:
-			    x-fabric-whitelist:
-			      service-list:
-			      - stups_spp-brand-service-loadtest
+				/api/brands/*:
+				  delete:
+				    x-fabric-whitelist:
+				      service-list:
+				      - stups_spp-brand-service-loadtest
 
-		*/
-		//return fmt.Errorf("invalid number of x-fabric-privileges %d", n)
-	}
+			//return fmt.Errorf("invalid number of x-fabric-privileges %d", n)
+		}
+	*/
 
 	if fm.Ratelimit != nil {
 		if fm.Ratelimit.DefaultRate < 1 {

--- a/fabric/fabrictest/fixtures.go
+++ b/fabric/fabrictest/fixtures.go
@@ -130,20 +130,6 @@ func safeFileClose(t *testing.T, fd *os.File) {
 	}
 }
 
-func compileRegexps(s []string) ([]*regexp.Regexp, error) {
-	var r []*regexp.Regexp
-	for _, si := range s {
-		ri, err := regexp.Compile(si)
-		if err != nil {
-			return nil, err
-		}
-
-		r = append(r, ri)
-	}
-
-	return r, nil
-}
-
 func testFixture(t *testing.T, f fixtureSet) {
 	// TODO(sszuecs): cleanup
 	//println("f.name:", f.name, "f.resources:", f.resources, "f.api:", f.api, "f.eskip:", f.eskip)

--- a/fabric/fabrictest/fixtures.go
+++ b/fabric/fabrictest/fixtures.go
@@ -30,17 +30,18 @@ type fixtureSet struct {
 	log            string
 }
 
-type kubeOptionsParser struct {
-	EastWest                 bool               `yaml:"eastWest"`
-	EastWestDomain           string             `yaml:"eastWestDomain"`
-	EastWestRangeDomains     []string           `yaml:"eastWestRangeDomains"`
-	EastWestRangePredicates  []*eskip.Predicate `yaml:"eastWestRangePredicatesAppend"`
-	HTTPSRedirect            bool               `yaml:"httpsRedirect"`
-	HTTPSRedirectCode        int                `yaml:"httpsRedirectCode"`
-	BackendNameTracingTag    bool               `yaml:"backendNameTracingTag"`
-	OnlyAllowedExternalNames bool               `yaml:"onlyAllowedExternalNames"`
-	AllowedExternalNames     []string           `yaml:"allowedExternalNames"`
-}
+// TODO(sszuecs): some of these might be features we need in case we do a PR to skipper
+// type kubeOptionsParser struct {
+// 	EastWest                 bool               `yaml:"eastWest"`
+// 	EastWestDomain           string             `yaml:"eastWestDomain"`
+// 	EastWestRangeDomains     []string           `yaml:"eastWestRangeDomains"`
+// 	EastWestRangePredicates  []*eskip.Predicate `yaml:"eastWestRangePredicatesAppend"`
+// 	HTTPSRedirect            bool               `yaml:"httpsRedirect"`
+// 	HTTPSRedirectCode        int                `yaml:"httpsRedirectCode"`
+// 	BackendNameTracingTag    bool               `yaml:"backendNameTracingTag"`
+// 	OnlyAllowedExternalNames bool               `yaml:"onlyAllowedExternalNames"`
+// 	AllowedExternalNames     []string           `yaml:"allowedExternalNames"`
+// }
 
 func baseNoExt(n string) string {
 	e := filepath.Ext(n)

